qiskit_machine_learning/connectors/torch_connector.py

```diff
@@ -232,6 +232,18 @@ def neural_network(self):
 
     @property
     def weights(self):
+        """Returns the weights of the underlying network."""
+        warnings.warn(
+            "The 'weights' property is deprecated as of 0.2.0 "
+            "and will be removed no sooner than 3 months after the release. "
+            "You should use the 'weight' property instead.",
+            DeprecationWarning,
+        )
+
+        return self._weights
+
+    @property
+    def weight(self):
```

Comment:
I guess its a choice, we could rename the private var to _weight to make things perhaps more consistent, though of course is not needed.

Should we add a return typehint here (I see in Torch Linear weight is of type Tensor) Same about nice to have a return type for sparse() below and neural_network() above which I happen to notice don't have it either

Reply:
We have `weights` everywhere in the code, so I prefer to keep the variable name as it is now. In the constructor I'd keep the parameter name as it is now as well. There's no a direct analogy of this constructor in pytorch. But `weight` instead of `weights` is a good change. In this case, if I have a large model with a QNN as one of the layers and I can inspect classical layers and a quantum one in the same fashion.
Type hints are added.